clang/lib/Parse/ParseDeclCXX.cpp

```diff
@@ -140,6 +140,11 @@ Parser::DeclGroupPtrTy Parser::ParseNamespace(DeclaratorContext Context,
       SkipUntil(tok::semi);
       return nullptr;
     }
+    if (!ExtraNSs.empty()) {
+      Diag(IdentLoc, diag::err_unexpected_qualified_namespace_alias);
```

Comment:
This is more of a cosmetic thing, but it would be nice to `<<` a `SourceRange` here that contains everything from the first `::` to the last identifier (e.g. in `namespace a::b::c = d`, that would be `::b::c`) so we underline it in the diagnostic, and maybe also use the location of the first `::` as the location for the diagnostic.